apm-sniffer/apm-sdk-plugin/hbase-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hbase/HTableInterceptor.java

```diff
@@ -0,0 +1,129 @@
+                if (!CollectionUtil.isEmpty(list) && list.get(0) instanceof OperationWithAttributes) {
+                    operations = list;
+                    canTracingServer = true;
+                }
+            } else if (allArguments[0] instanceof OperationWithAttributes) {
+                operation = (OperationWithAttributes) allArguments[0];
+                canTracingServer = true;
+            }
+        }
+        AbstractSpan span;
+        if (canTracingServer) {
+            ContextCarrier contextCarrier = new ContextCarrier();
+            span = ContextManager.createExitSpan(PREFIX_OPERATION_NAME + method.getName(),
+                    contextCarrier, (String) objInst.getSkyWalkingDynamicField());
+            CarrierItem next = contextCarrier.items();
+            while (next.hasNext()) {
+                next = next.next();
+                if (operation != null) {
+                    operation.setAttribute(next.getHeadKey(), next.getHeadValue().getBytes());
+                } else {
+                    for (OperationWithAttributes o : operations) {
+                        o.setAttribute(next.getHeadKey(), next.getHeadValue().getBytes());
+                    }
+                }
+            }
+        } else {
+            span = ContextManager.createExitSpan(PREFIX_OPERATION_NAME + method.getName(),
+                    (String) objInst.getSkyWalkingDynamicField());
+        }
+        span.setComponent(ComponentsDefine.HBASE);
+        Tags.DB_TYPE.set(span, HBASE_DB_TYPE);
+        Tags.DB_INSTANCE.set(span, ((HTable) objInst).getName().getNameAsString());
+        SpanLayer.asDB(span);
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                              Object ret) throws Throwable {
+        ContextManager.stopSpan();
+        return ret;
+    }
+
+    @Override
+    public void handleMethodException(EnhancedInstance objInst, Method method, Object[] allArguments,
+                                      Class<?>[] argumentsTypes, Throwable t) {
+        AbstractSpan span = ContextManager.activeSpan();
+        span.errorOccurred();
+        span.log(t);
+    }
+
+    @Override
+    @SuppressWarnings("rawtypes")
+    public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
+        try {
+            Configuration connection = ((ClusterConnection) allArguments[1]).getConfiguration();
+            Field field = connection.getClass().getDeclaredField("overlay");
+            field.setAccessible(true);
+            Properties properties = (Properties) field.get(connection);
+            for (Map.Entry entry : properties.entrySet()) {
+                if ("hbase.zookeeper.quorum".equals(entry.getKey())) {
+                    objInst.setSkyWalkingDynamicField(entry.getValue().toString());
+                }
+            }
+        } catch (Exception e) {
```

Comment:
The agent core has `try-catch` already. We don't need this in the interceptor.

Reply:
just throw?